src/containers/NetworkViewers/RelationshipTreeViewer.tsx

```diff
@@ -440,26 +543,70 @@ class TreeViewer extends Component<Props, State> {
         });
       }
     });
+    const { filters } = this;
+    const nodeFilters = query
+      .map(el => {
+        const split = el.split('.');
+        if (!filters[split[0]] || !filters[split[0]][split[1]]) {
+          return undefined;
+        }
+        return filters[split[0]][split[1]].filterNode;
+      })
+      .filter(el => !!el) as ((node: RelationshipResource) => boolean)[];
+    const linkFilters = query
+      .map(el => {
+        const split = el.split('.');
+        if (!filters[split[0]] || !filters[split[0]][split[1]]) {
+          return undefined;
+        }
+        return filters[split[0]][split[1]].filterLink;
+      })
+      .filter(el => !!el) as ((link: Relationship) => boolean)[];
     Array.from(relationshipIds)
       .map(id => items[id])
       .forEach((relationship: Relationship) => {
-        nodes[relationship.source.resourceId] = {
-          ...relationship.source,
-          id: relationship.source.resourceId,
-          color: this.chooseNodeColor(relationship.source),
-        };
-        nodes[relationship.target.resourceId] = {
-          ...relationship.target,
-          id: relationship.target.resourceId,
-          color: this.chooseNodeColor(relationship.target),
-        };
-        if (relationship.source.resourceId !== relationship.target.resourceId) {
-          links.push({
-            ...relationship,
-            linkWidth: 3,
-            source: relationship.source.resourceId,
-            target: relationship.target.resourceId,
-          });
+        const sourcePassFilter = nodeFilters.some(filter =>
+          filter(relationship.source)
+        );
+        const targetPassFilter = nodeFilters.some(filter =>
+          filter(relationship.target)
+        );
+        if (
```

Comment:
This is extremely hard to read, and would like to see a small comment to describe what's going on here.